test/device_map_dialog_test.rb

```diff
@@ -0,0 +1,69 @@
+require_relative "test_helper"
+
+require "bootloader/device_map_dialog"
+
+describe Bootloader::DeviceMapDialog do
+  # just simple tests to avoid typos as logic is not so easy to test
+  describe "#run" do
+    before do
+      Yast.import "BootStorage"
+
+      device_map = Bootloader::DeviceMap.new(
+        "/dev/sda" => "hd0",
+        "/dev/sdb" => "hd1"
+      )
+      allow(Yast::BootStorage).to receive(:DisksOrder)
+        .and_return(device_map.disks_order)
+    end
+
+    def mock_ui_events(*events)
```

Comment:
I would reduce this method to just the first line. Putting the rest directly in the before block including some explanation for the QueryWidget one, which is not that obvious:
```ruby
# On every iteration, the event loop checks which disk is selected in the UI
allow(Yast::UI).to receive(:QueryWidget).and_return("/dev/sda")
```